geopandas/tests/test_geodataframe.py

```diff
@@ -1289,15 +1289,48 @@ def test_multiindex_with_geometry_label(self, dtype):
         df.columns = pd.MultiIndex.from_product([["geometry"], [0, 1]])
         # don't error in constructor
         gdf = GeoDataFrame(df)
-        # Getting the .geometry column gives GeoDataFrame for both columns
-        # (but with first MultiIndex level removed)
-        # TODO should this give an error instead?
-        result = gdf.geometry
-        assert result.shape == gdf.shape
-        assert result.columns.tolist() == [0, 1]
-        assert_frame_equal(result, gdf["geometry"])
-        result = gdf[["geometry"]]
-        assert_frame_equal(result, gdf if dtype == "geometry" else pd.DataFrame(gdf))
+        with pytest.raises(AttributeError, match=".*geometry .* has not been set.*"):
```

Comment:
I made the assertion this should error now, in this case where gdf["geometry"] matches multiple columns and returns a dataframe I think it's reasonable that gdf.geometry raises, I think it's reasonable to expect that gdf.geometry is always a GeoSeries.

I'm just not sure what else should actually be tested here.

Reply:
> I think it's reasonable that gdf.geometry raises, I think it's reasonable to expect that gdf.geometry is always a GeoSeries.

Yes, I would also have that expectation (you can use `gdf["geometry"]` if you want the "standard" pandas behaviour)